lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -998,6 +998,41 @@ SmallVector<unsigned> DotOperandEncodingAttr::getShapePerCTATile(
   }
 }
 
+LogicalResult DotOperandEncodingAttr::verify(
+    ::llvm::function_ref<::mlir::InFlightDiagnostic()> emitError,
+    unsigned opIdx, Attribute parent, unsigned kWidth) {
+  if (opIdx != 0 && opIdx != 1)
+    return emitError() << "DotOperandEncoding::opIdx can be 0 or 1, got "
+                       << opIdx;
+  if (!parent)
+    return emitError()
+           << "DotOperandEncoding::parent is manadatory parameter, got nullptr";
+  if (kWidth != 0) {
+    // general confines on parent layouts
+    if (!parent.isa<NvidiaMmaEncodingAttr, AMDWmmaEncodingAttr,
+                    AMDMfmaEncodingAttr>())
+      return emitError()
+             << "DotOperandEncoding::kWidth is not supported for given parent: "
+             << parent;
+    // additional confines
+    if (parent.isa<NvidiaMmaEncodingAttr>() &&
+        !parent.cast<NvidiaMmaEncodingAttr>().isAmpere())
+      return emitError()
+             << "DotOperandEncoding::kWidth is non zero only for Ampere";
```

Comment:
Not sure the error message parses for me.. You can ".. kWidth can only be non-zero for Ampere architecture"?